### PR TITLE
[Docs] Shannon Governance Parameter Documentation Page

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

- Added a small utility `make docs_update_gov_params_page`
- Generates the documentation for all of our governance parms at `docusaurus/docs/protocol/governance/params.md` governance page
- Autoamtically read the comments and details in the `params.proto` files in the source code

Screenshot 2024-08-15 at 3 07 22 PM

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [x] Documentation
- [ ] Other (specify)

## Testing

- [x] `make docusaurus_start`; only needed if you make doc changes

## Summary by CodeRabbit

- **New Features**
  - Introduced a new target for updating governance-related documentation, enhancing the automation process.
  - Added a comprehensive documentation page outlining governance parameters for the Pocket Network.
  - Implemented an automated workflow for generating and updating governance documentation.

- **Documentation**
  - Minor formatting adjustments made for clarity in existing documentation.
  - Added clear guidelines for updating governance parameters to ensure consistency.

### CODE REVIEW

File: tools/scripts/generate_docs_params.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+			return "", err
+		}
+		defer protoFile.Close()
+
+		var paramsMessages []ProtoMessage
+		var currentParamMessage *ProtoMessage
+		var currentComment string
+
+		messageParamPattern := regexp.MustCompile(`^message\s+(Params)\s*{`)
+		fieldPattern := regexp.MustCompile(`^\s*(\w+)\s+(\w+)\s*=\s*(\d+)\s*\[(.*?)\];`)
```
Comment (red-0ne) [inline tools/scripts/generate_docs_params.go:118]: Could we have these regexp patterns outside this function as top level vars?

File: tools/scripts/generate_docs_params.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+				continue
+			}
+
+			// Check if the line defines a new message
+			if matches := messageParamPattern.FindStringSubmatch(line); matches != nil {
+				if currentParamMessage != nil {
+					paramsMessages = append(paramsMessages, *currentParamMessage)
+				}
+				currentParamMessage = &ProtoMessage{Name: matches[1]}
+				continue
```
Comment (red-0ne) [inline tools/scripts/generate_docs_params.go:137]: We should reset any comment seen before seeing `message Params` for the first time.

**Review by red-0ne — APPROVED** [review]
This is great! Preemptively approving but I left a couple of comments there.

Also, could we have this as part of our CI instead of manually generating it?

File: tools/scripts/generate_docs_params.go (hunk 2 above)
Comment (Olshansk) [reply tools/scripts/generate_docs_params.go:137]: This caught a bug!

File: .github/workflows/upload-pages-artifact.yml [hunk 3]
```diff
@@ -13,6 +13,21 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  update-docs:
```
Comment (Olshansk) [inline .github/workflows/upload-pages-artifact.yml:16]: @okdas Could you please review this one part.

**Review by okdas — APPROVED** [review]
Looks good, go code seems to be working too. I did not look into the go code itself, though! Let me know if you want me to.

Screenshot 2024-08-16 at 4 19 27 PM

**Pull request merged**